### DISCUSSION TITLE
Which apollo-server variant should be used on API routes?

### DISCUSSION THREAD

**Discussion by jstcki** [vercel/next.js#11399] [Help]
In the [api-routes-graphql](https://github.com/zeit/next.js/tree/canary/examples/api-routes-graphql) example, `apollo-server-micro` is used. I have also implemented this and it works perfectly fine as far as I can tell.

However, recently the use of micro middlewares was discouraged in favor of using [Connect/Express middlewares](https://nextjs.org/docs/api-routes/api-middlewares#connectexpress-middleware-support), which adds the need to write some boilerplate code to promisify express-style middleware.

Now, I'm wondering if in above's example `apollo-server-express` should be used or if there is a better solution?

**Comment by timneutkens** [comment]
Basically people were confusing semantics of Micro they're slightly different than API routes, and thought Next.js used Micro as they were both created by ZEIT. However API routes do not use Micro.

apollo-server-micro probably works fine as it doesn't do anything special with `req` and `res`
